content/news/2021-08-19-gsoc-coverage.md

```diff
@@ -0,0 +1,130 @@
+realized I would encounter the exact same problem as I had with Secret Storage.
+This MSC introduced the event type `*m.notifications_profile.<profile>*` where
+`<profile>` is the name of the profile to store. I still powered ahead,
+implementing the different endpoints and events, temporarily setting a fixed
+event type name for `*m.notifications_profile.<profile>`.  This allowed the
+grunt of the work to be done, meaning it would be ready for when Ruma gets
+support for variable event names. Although the MSC isn't finalized as of yet,
+all the currently laid out endpoints and events have been implemented in a
+branch ready to be merged once possible.
+
+Thus there are a couple of pull requests which are blocked and that will be
+merged in the future, once variable event type names are supported:
+
+- [Add support for MSC2785: Event notification attributes and actions]
+- [Add secret storage]
+
+Finally, I worked on a final issue for my GSoC project:
+
+- [Generate for impls for event enums]
+
+To solve this last issue, I decided to learn how to write procedural macros. In
+the first half of GSoC, I had created a new `macro_rules!` macro, largely based
+on a pre-existing one. As I had very little experience in writing macros, and
+zero experience in writing procedural ones, I decided to read a short series of
+[blog posts] to get myself up to speed. Once that was done, I got to work: I
+added the generation of `From` trait implementations inside the pre-existing
+`[proc-macro]` functions which generate the event enums. I also added a `Derive`
+macro for enums created manually. I really enjoyed working on this, which was my
+first foray into Rust's procedural macros.
+
+## Recap of PRs for my GSoC
+
+### Start of GSoC with the Identity Service API
+- [3PID validity endpoint]
+- [3PID bind endpoint]
+- [Invitation signing endpoint]
+- [Store invitation endpoint]
+- [3PID unbind endpoint]\(finished at the end of GSoC)
+
+### Middle of GSoc, implementing MSCs and adding types
+
+- [Add client secret and session identifier types]
+- [Add RoomName struct]
+- [Update endpoints for Blurhash implementation]
+- [Add "knock" feature from MSC2403]
+- [Implement reasons for leaving a room]
+
+### End of GSoC
+
+- [Add support for MSC2785: Event notification attributes and actions]  
+    \(commit `c556e1c5cb1894a6a357799557b07088deb6bea3`)
+- [Add secret storage]  
+    \(commit `4eae28643036321c7ad86dc8d887b5790b68092b`)
+- [Generate for impls for event enums]
+
+
+
+[previous blog post]: https://www.ruma.io/news/gsoc-2021-intro/
+[3PID validity endpoint]: https://github.com/ruma/ruma/pull/618
+[3PID bind endpoint]: https://github.com/ruma/ruma/pull/621
+[Invitation signing endpoint]: https://github.com/ruma/ruma/pull/626
+[Store invitation endpoint]: https://github.com/ruma/ruma/pull/631
+[3PID unbind endpoint]: https://github.com/ruma/ruma/pull/627
+[Add client secret and session identifier types]: eat
```

Comment:
What happened here? 😄

Reply:
I think I had clicked "copy" instead of "copy link", and had "eat" selected randomly somewhere. I guess this one slipped past :sweat_smile: